src/components/NcAppSidebar/NcAppSidebarTabs.vue

```diff
@@ -192,89 +111,176 @@ export default {
 		 * and emit to the parent component
 		 */
 		focusLastTab() {
-			this.setActive(this.tabs[this.tabs.length - 1].id)
+			this.setActive(this.tabs[this.tabs.length - 1].componentOptions.propsData.id)
 			this.focusActiveTab() // focus nav item
 		},
 
 		/**
 		 * Focus the current active tab
 		 */
 		focusActiveTab() {
-			this.$el.querySelector('#' + this.activeTab).focus()
+			// We have to wait until the component finished rendering
+			this.$nextTick(
+				() => this.$el.querySelector('#' + this.active).focus()
+			)
 		},
 
 		/**
 		 * Focus the content on tab
 		 * see aria accessibility guidelines
 		 */
 		focusActiveTabContent() {
-			this.$el.querySelector('#tab-' + this.activeTab).focus()
+			// We have to wait until the component finished rendering
+			this.$nextTick(
+				() => this.$el.querySelector('#tab-' + this.active).focus()
+			)
 		},
 
-		/**
-		 * Update the current active tab
-		 */
-		updateActive() {
-			this.activeTab = this.active
-				&& this.tabs.findIndex(tab => tab.id === this.active) !== -1
-				? this.active
-				: this.tabs.length > 0
-					? this.tabs[0].id
-					: ''
-		},
-
-		hasMdIcon(tab) {
-			return tab?.$slots?.icon
+		onKeydown($event) {
+			$event.preventDefault()
+			// Ensure the key event is 'exact'
+			if (['ctrl', 'shift', 'alt', 'meta'].some(modifier => $event[`${modifier}Key`])) {
+				return
+			}
+			// Left
+			if ($event.keyCode === 37) {
+				this.focusPreviousTab()
+			}
+			// Right
+			if ($event.keyCode === 39) {
+				this.focusNextTab()
+			}
+			// Tab
+			if ($event.keyCode === 9) {
+				this.focusActiveTabContent()
+			}
+			// Page-Up
+			if ($event.keyCode === 33) {
+				this.focusFirstTab()
+			}
+			// Page-Down
+			if ($event.keyCode === 34) {
+				this.focusLastTab()
+			}
 		},
+	},
 
+	/**
+	 * The render function to display the component
+	 *
+	 * @param {Function} h The function to create VNodes
+	 * @return {object|undefined} The created VNode
+	 */
+	render(h) {
 		/**
-		 * Manually update the sidebar tabs according to $slots.default
+		 * Follows the tab aria guidelines
+		 * https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html
 		 */
-		updateTabs() {
-			if (!this.$slots.default) {
-				this.tabs = []
-				return
+		 if (!this.$slots.default) {
+			this.tabs = []
+			return
+		}
```

Comment:
This is a rendering side-effect.

Reply:
> This is a rendering side-effect.

Depends a bit. `this.tabs` is not used in the render function itself, it's just necessary to activate the correct tab when using the keyboard.